store/tikv/region_request.go

```diff
@@ -428,14 +428,23 @@ func (s *RegionRequestSender) sendReqToRegion(bo *Backoffer, rpcCtx *RPCContext,
 	}
 
 	injectFailOnSend := false
-	if connID > 0 {
-		failpoint.Inject("rpcFailOnSend", func() {
-			logutil.Logger(ctx).Info("[failpoint] injected RPC error on send", zap.Stringer("type", req.Type),
-				zap.Stringer("req", req.Req.(fmt.Stringer)), zap.Stringer("ctx", &req.Context))
-			injectFailOnSend = true
-			err = errors.New("injected RPC error on send")
-		})
-	}
+	failpoint.Inject("rpcFailOnSend", func(val failpoint.Value) {
+		// Optional filters
+		if s, ok := val.(string); ok {
+			if s == "greengc" && !req.IsGreenGCRequest() {
+				return
+			} else if s == "write" && !req.IsTxnWriteRequest() {
+				return
+			}
+		} else if connID == 0 {
+			return
+		}
```

Comment:
Here return is actually the return of sendReqToRegion after failpoint rewriting.

Reply:
Oh... that's wierd...